pymeasure/instruments/keithley/keithley2400.py

```diff
@@ -303,6 +303,83 @@ class Keithley2400(Instrument, KeithleyBuffer):
         values=[0, 999.9999]
     )
 
+    ###########
+    # Filters #
+    ###########
+
+    current_filter_type = Instrument.control(
+        ":SENS:CURR:AVER:TCON?", ":SENS:CURR:AVER:TCON %s",
+        """ A String property that controls the filter's type for the current.
+        REP : Repeating filter
+        MOV : Moving filter""",
+        validator=strict_discrete_set,
+        values=['REP', 'MOV'],
+        map_values=False)
+
+    current_filter_count = Instrument.control(
+        ":SENS:CURR:AVER:COUNT?", ":SENS:CURR:AVER:COUNT %d",
+        """ A integer property that controls the number of readings that are 
+        acquired and stored in the filter buffer for the averaging""",
+        validator=truncated_range,
+        values=[1, 2500],
+        cast=int)
+
+    current_filter_state = Instrument.control(
+        ":SENS:CURR:AVER?", ":SENS:CURR:AVER %s",
+        """ A string property that controls if the filter is active.""",
+        validator=strict_discrete_set,
+        values=['ON', 'OFF'],
+        map_values=False)
+
+    voltage_filter_type = Instrument.control(
+        ":SENS:VOLT:AVER:TCON?", ":SENS:VOLT:AVER:TCON %s",
+        """ A String property that controls the filter's type for the current.
+        REP : Repeating filter
+        MOV : Moving filter""",
+        validator=strict_discrete_set,
+        values=['REP', 'MOV'],
+        map_values=False)
+
+    voltage_filter_count = Instrument.control(
+        ":SENS:VOLT:AVER:COUNT?", ":SENS:VOLT:AVER:COUNT %d",
+        """ A integer property that controls the number of readings that are 
+        acquired and stored in the filter buffer for the averaging""",
+        validator=truncated_range,
+        values=[1, 2500],
+        cast=int)
+
+    #####################
+    # Output subsystem #
+    #####################
+
+    current_output_off_state = Instrument.control(
+        ":OUTP:CURR:SMOD?", ":OUTP:CURR:SMOD %s",
+        """ Select the output-off state of the SourceMeter.
+        HIMP : output relay is open, disconnects external circuitry.
+        NORM : V-Source is selected and set to 0V, Compliance is set to 0.5% 
```

Comment:
Is the meaning of the various parameters really identical between current_ and voltage_ versions?

Reply:
The documentation is quite shallow on this function. I added it after someone requested it (see issue #319). From what I understand, depending on which output is currently selected, the Keithley will switch to right kind of off state mode. I don't know how to test it though to confirm the behavior.

Reply:
OK then.